aictl/cli/main.py

```diff
@@ -189,6 +189,52 @@ def t2a(args):
     )
 
 
+def upscale(args):
+    import torch
+    from RealESRGAN import RealESRGAN
+    from diffusers import StableDiffusionUpscalePipeline
+
+    # check if on mac and mps is available, fallback to cuda then cpu
+    is_mac = False
+    device = "cuda:0" if torch.cuda.is_available() else "cpu"
+    if torch.backends.mps.is_available():
+        is_mac = True
+        device = torch.device("mps")
+        print("MPS device detected. Using MPS.")
+
+    # Get the image
+    if args.image is None:
+        image = download_image(args.image_url)
+    else:
+        image = load_image_from_path(args.image)
+    # SDX4 Upscaler
+    if args.model == "sdx4":
+        # load models and configure pipeline settings
+        print("### loading models")
+        model_type = torch.float32 if is_mac else torch.float16
+        model_id = "stabilityai/stable-diffusion-x4-upscaler"
+        pipe = StableDiffusionUpscalePipeline.from_pretrained(
+            model_id, torch_dtype=model_type
+        )
+        pipe.scheduler = args.scheduler.from_config(pipe.scheduler.config)
+        if is_mac:
+            pipe.enable_attention_slicing()
+        else:
+            pipe.enable_model_cpu_offload()
+            pipe.enable_xformers_memory_efficient_attention()
+        pipe = pipe.to(device)
+        upscaled_image = pipe(prompt=args.prompt, image=image).images[0]
+        upscaled_image.save(args.output_path)
+
+    if args.model == "esrgan":
```

Comment:
I know it won't make a difference but logically speaking this should be an `elif`

Reply:
done